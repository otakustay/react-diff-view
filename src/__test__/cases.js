/* In case you are looking for type of hunk, see src/Hunk/index.js */
import {parseDiff} from '..';

export const basic = `diff --git a/src/__test__/index.test.jsx b/src/__test__/index.test.jsx
index 643c2f0..7883597 100644
--- a/src/__test__/index.test.jsx
+++ b/src/__test__/index.test.jsx
@@ -21,3 +21,3 @@ describe('basic test', () => {
     test('App renders correctly', () => {
-        expect(renderer.create(<App diffText={'deff'} />).toJSON()).toMatchSnapshot();
+        expect(renderer.create(<App diffText={'diff'} />).toJSON()).toMatchSnapshot();
     });`;

export const basicHunk = parseDiff(basic)[0].hunks[0];

export const multiple = `diff --git a/src/index1.js b/src/index1.js
index e69de29..d00491f 100644
--- a/src/index1.js
+++ b/src/index1.js
@@ -0,0 +1 @@
+1
diff --git a/src/index.js b/src/index2.js
index d00491f..0cfbf08 100644
--- a/src/index2.js
+++ b/src/index2.js
@@ -1 +1 @@
-1
+2
diff --git a/src/index3.js b/src/index3.js
index 0cfbf08..e69de29 100644
--- a/src/index3.js
+++ b/src/index3.js
@@ -1 +0,0 @@
-2`;

export const multipleHunk = parseDiff(basic)[0].hunks[0];
