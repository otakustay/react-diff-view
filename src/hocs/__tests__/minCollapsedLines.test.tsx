import * as renderer from 'react-test-renderer';
import {basicHunk} from '../../__tests__/cases';
import {minCollapsedLines} from '..';

function ComponentIn() {
    return <div />;
}

describe('minCollapsedLines', () => {
    test('returns a hoc function', () => {
        expect(typeof minCollapsedLines(0)).toBe('function');
    });

    test('hoc function return a component', () => {
        expect(typeof minCollapsedLines(0)(ComponentIn)).toBe('function');
    });

    test('render component', () => {
        const ComponentOut = minCollapsedLines(0)(ComponentIn);
        expect(renderer.create(<ComponentOut />).toJSON()).toMatchSnapshot();
    });

    test('render component with minLinesExclusive', () => {
        const ComponentOut = minCollapsedLines(10)(ComponentIn);
        expect(renderer.create(<ComponentOut />).toJSON()).toMatchSnapshot();
    });

    test('render component with oldSource', () => {
        const ComponentOut = minCollapsedLines(10)(ComponentIn);
        expect(renderer.create(<ComponentOut oldSource hunks={[basicHunk]} />).toJSON()).toMatchSnapshot();
    });
});
