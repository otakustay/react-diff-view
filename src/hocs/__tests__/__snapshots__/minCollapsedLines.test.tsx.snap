// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`minCollapsedLines render component 1`] = `<div />`;

exports[`minCollapsedLines render component with minLinesExclusive 1`] = `<div />`;

exports[`minCollapsedLines render component with oldSource 1`] = `<div />`;
